pygmt/src/select.py

```diff
@@ -0,0 +1,146 @@
+"""
+select - Select data table subsets based on multiple spatial criteria.
+"""
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    A="area_thresh",
+    I="reverse",
+    J="projection",
+    R="region",
+    V="verbose",
+    Z="z_subregion",
+    b="binary",
+    d="nodata",
+    e="find",
+    f="coltypes",
+    g="gap",
+    h="header",
+    i="incols",
+    o="outcols",
+    r="registration",
+    s="skiprows",
+    w="wrap",
+)
+@kwargs_to_strings(R="sequence", i="sequence_comma", o="sequence_comma")
+def select(data=None, outfile=None, **kwargs):
+    r"""
+    Select data table subsets based on multiple spatial criteria.
+
+    This is a filter that reads (x, y) or (longitude, latitude) positions from
+    the first 2 columns of *data* and uses a combination of 1-7 criteria to
+    pass or reject the records. Records can be selected based on whether or not
+    they are:
+
+    1. inside a rectangular region (**region** [and **projection**])
+    2. within *dist* km of any point in *pointfile*
+    3. within *dist* km of any line in *linefile*
+    4. inside one of the polygons in the *polygonfile*
+    5. inside geographical features (based on coastlines)
+    6. has z-values within a given range, or
+    7. inside bins of a grid mask whose nodes are non-zero
+
+    The sense of the tests can be reversed for each of these 7 criteria by
+    using the **reverse** option.
+
+    Full option list at :gmt-docs:`gmtselect.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    data : str or {table-like}
+        Pass in either a file name to an ASCII data table, a 2D
+        {table-classes}.
```

Comment:
```suggestion
        Pass in either a file name to an ASCII data table or a 2D
        {table-classes}.
```

I"m not familiar with using `gmtselect`, but can't it accept 3D tables (as indicated by `z_subregion`?